core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCommonService.java

```diff
@@ -142,19 +141,17 @@ private FlowIdentifier clusterHostgroupAdjustmentChange(Long stackId, UpdateClus
         }
         LOGGER.debug("Cluster host adjustment request received. Stack id: {} ", stackId);
         Blueprint blueprint = stack.getCluster().getBlueprint();
-        Optional<HostGroup> hostGroup = hostGroupService.findHostGroupInClusterByName(stack.getCluster().getId(),
-                updateJson.getHostGroupAdjustment().getHostGroup());
-        if (hostGroup.isEmpty()) {
-            throw new BadRequestException(String.format("Host group '%s' not found or not member of the cluster '%s'",
-                    updateJson.getHostGroupAdjustment().getHostGroup(), stack.getName()));
+        String hostGroupName = updateJson.getHostGroupAdjustment().getHostGroup();
+        if (hostGroupService.hasHostGroupInCluster(stack.getCluster().getId(), hostGroupName)) {
+            throw new BadRequestException(String.format("Host group '%s' not found or not member of the cluster '%s'", hostGroupName, stack.getName()));
```

Comment:
Can we have an additional log here? Sry, it is not strictly related, but would be beneficial here.

Reply:
The exception mapper will log the message and stack trace anyway. Does it make sense to log it here as well?

Reply:
Yes, it could work in case of a badrequest exception.